sdk/textanalytics/ai-text-analytics/src/util.ts

```diff
@@ -32,7 +32,14 @@ export function sortResponseIdObjects<T extends IdObject, U extends IdObject>(
     );
   }
 
-  return sortedArray.map((item) => unsortedMap.get(item.id)!);
+  let result: U[] = [];
+  for (const sortedItem of sortedArray) {
+    const item = unsortedMap.get(sortedItem.id);
+    if (item) {
```

Comment:
because of pagination, the sizes of the two arrays is not the same, the array of the input documents  can be bigger than the size of the current page of results.